sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -175,19 +187,71 @@ def test_check_estimator():
 
     # doesn't error on actual estimator
     check_estimator(AdaBoostClassifier)
+    check_estimator(AdaBoostClassifier())
     check_estimator(MultiTaskElasticNet)
+    check_estimator(MultiTaskElasticNet())
+
+
+def _bad_fit(self, X, y):
+    raise ValueError("You shouldn't have called this")
+
+
+def test_check_estimator_clones():
+    # check that check_estimator doesn't modify the estimator it receives
+    from sklearn.datasets import load_iris
+    iris = load_iris()
+
+    for name, Estimator in all_estimators():
+        est = Estimator()
+        set_checking_parameters(est)
+        # without fitting
+        est.fit = _bad_fit
+        old_pickle = pickle.dumps(est)
+        try:
+            check_estimator(est)
+        except Exception as e:
+            # some estimators don't pass the test right now
+            # don't worry about that here
+            if "called this" in str(e):
```

Comment:
Maybe putting the whole message that you expect here would be more clear, ie "You shouldn't have called this"